legal-api/src/legal_api/models/legal_entity.py

```diff
@@ -588,6 +588,10 @@ def find_by_internal_id(cls, internal_id: int = None):
         legal_entity = None
         if internal_id:
             legal_entity = cls.query.filter_by(id=internal_id).one_or_none()
+        if not legal_entity:
+            return None
+        if legal_entity.alternate_name_entity:
+            return legal_entity.alternate_name_entity
```

Comment:
I don't think we want to update `LegalEntity.find_by_internal_id` like this.  You need have access to the right business(AlternateName or LegalEntity) before you call the `find_by_internal_id` function